public/components/datasources/components/new/configure_prometheus_datasource.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useState } from 'react';
+import { access } from 'fs';
+import { OPENSEARCH_DOCUMENTATION_URL } from '../../../../../common/constants/data_connections';
+import { QueryPermissionsConfiguration } from '../manage/query_permissions';
+import { Role } from '../../../../../common/types/data_connections';
+
+interface ConfigurePrometheusDatasourceProps {
+  roles: Role[];
+  selectedQueryPermissionRoles: Role[];
+  setSelectedQueryPermissionRoles: React.Dispatch<React.SetStateAction<Role[]>>;
+  currentName: string;
+  currentDetails: string;
+  currentStore: string;
+  currentUsername: string;
+  currentPassword: string;
+  currentAccessKey: string;
+  currentSecretKey: string;
+  currentRegion: string;
+  setRegionForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setAccessKeyForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setSecretKeyForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setPasswordForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setUsernameForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setStoreForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setNameForRequest: React.Dispatch<React.SetStateAction<string>>;
+  setDetailsForRequest: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const ConfigurePrometheusDatasource = (props: ConfigurePrometheusDatasourceProps) => {
+  const {
+    setNameForRequest,
+    setDetailsForRequest,
+    setStoreForRequest,
+    currentStore,
+    currentName,
+    currentDetails,
+    roles,
+    selectedQueryPermissionRoles,
+    setSelectedQueryPermissionRoles,
+    currentUsername,
+    setUsernameForRequest,
+    currentPassword,
+    setPasswordForRequest,
+    currentAccessKey,
+    setAccessKeyForRequest,
+    currentSecretKey,
+    setSecretKeyForRequest,
+    currentRegion,
+    setRegionForRequest,
+  } = props;
+
+  const [name, setName] = useState(currentName);
+  const [details, setDetails] = useState(currentDetails);
+  const [store, setStore] = useState(currentStore);
+  const authOptions = [
+    { value: 'basic', text: 'Basic Auth' },
+    { value: 'sigv4', text: 'SIGV4' },
+  ];
+  const [selectedAuthOption, setSelectedAuthOption] = useState(authOptions[0].value);
+
+  const AuthDetails = () => {
+    const [password, setPassword] = useState(currentPassword);
+    const [username, setUsername] = useState(currentUsername);
+    const [accessKey, setAccessKey] = useState(currentAccessKey);
+    const [secretKey, setSecretKey] = useState(currentSecretKey);
+    const [region, setRegion] = useState(currentRegion);
+    switch (selectedAuthOption) {
+      case 'basic':
+        return (
+          <>
+            <EuiFormRow label="Username">
+              <EuiFieldText
+                placeholder={'Username placeholder'}
+                value={username}
+                onChange={(e) => setUsername(e.target.value)}
+                onBlur={(e) => setUsernameForRequest(e.target.value)}
```

Comment:
@joshuali925 @ps48 was trying to figure a way around all this boiler plate code - issue is that when form is in a child component, re-rendering is causing it to lose focus when input changes. Am working around it with onBlur and temp var. in child component, which defaults to parent variable on load - is there a better way to do this?

Reply:
What is the cause of re-rendering? Suspecting: Should the switch statement be in a useEffect()?

Reply:
What I am thinking - I am passing down the value for that field (for example to persist across navigating to different part of the form and coming back). However, when the child component's onChange is triggered, this updates the parent's value of the variable. This then causes all the children to re-render, and the field loses focus, since it isn't focused by default. The way I am working around this for now/in this PR is to only trigger the parent's update `setValue` hook `onBlur` (i.e. when the user stops typing into the field). However, this is causing a lot of boiler plate with needing another temporary value in the child component. Is there a cleaner way around this?